### ISSUE TITLE
please use composer

### ISSUE THREAD

**Issue by alihammad-gist** [trippo/ResponsiveFilemanager#136]
please use composer

**Comment by garak** [comment]
:+1:

**Comment by joe936** [comment]
Yes, also, some instructions would be nice too. I have no idea how to configure this bundle.

**Comment by pkalisz** [comment]
@trippo , how do you anticipate, when this rewritten version would be available? I would really want to use it with my Symfony project, but in current shape it is quite hard to achieve.

**Comment by rubenheymans** [comment]
I would like to know to. And is it possible to make this project public on github, as an alpha/beta version, so we can help?

**Comment by mark9000** [comment]
This is a very important issue for me too. And the config is not really a problem. For example: I have a post-update-cmd in my root composer.json which writes my individual config file. And i think the composer file will not break anything. Thanks!

**Comment by jarodium** [comment]
Pre install scripts on composer can be used to make a copy of an older file to a new one.

**Comment by Kingtreemonkey** [comment]
I've done a pretty much complete re-write of this to suit a silex based application, could do with a tidy up but most functionality is working.

**Comment by rubenheymans** [comment]
@trippo is it possible to implement this?

**Comment by Jbaal** [comment]
+1 for the usage of composer
I try to integrate it to my new Symfony project, @Kingtreemonkey could you at least create a new repository in Github to have a look at your rewrite ?

**Comment by Kingtreemonkey** [comment]
sure, could do with a bit of tidying up in parts - will sort this evening (time permitting!)

**Comment by Jbaal** [comment]
Thanks a lot :)

**Comment by Kingtreemonkey** [comment]
@Jbaal https://github.com/Kingtreemonkey/FileManager Hope it's of help!

**Comment by Jbaal** [comment]
@Kingtreemonkey Thanx again, really helpfull ;)

**Comment by garak** [comment]
any news?

**Comment by rubenheymans** [comment]
@Kingtreemonkey Are you planning on updating this any time soon?

**Comment by Kingtreemonkey** [comment]
Hi Ruben,

If you have any questions on my version / request(s) please enter them into
an issue on https://github.com/Kingtreemonkey/FileManager

Thanks!

**Comment by gordon-matt** [comment]
I think this would not be a good idea. I am a .NET developer and have managed to get this file manager working in ASP.NET MVC (See: http://www.codeproject.com/Tips/1004596/Using-Responsive-File-Manager-in-ASP-NET-MVC). I don't know much about PHP really.. but for me, keeping it as simple PHP is fine, but making it only work with some PHP framework would make future versions unusable.

No, I think the way forward is not to rewrite the file manager to be framework specific, but rather to make it pure HTML5/JavaScript. That way, all web developers can make use of this.. regardless of whether they're using PHP, .NET, Java, Ruby, etc..

**Comment by SamMousa** [comment]
@gordon-matt Composer is not a framework; it is a dependency manager.
I'm all for separating backend (PHP) from frontend (HTML / JS).

But as long as there is PHP code and there are dependencies on other libraries composer is the way to go.

If backend and frontend are separated it makes sense to use something like Bower for the javascript dependencies and composer for the PHP backend.

**Comment by gordon-matt** [comment]
@SamMousa : I rely on a PHP compiler called Phalanger. As long as using this "Composer" doesn't mean incompatibility for me, sure.. I have no issue there.  :-)

However, I have noted that Alberto has on several occasions mentioned that he is looking to rewrite the whole thing, using AngularJS. I still think that is the best way forward: pure JavaScript. It would also mean more of us can more easily make use of and even contibute to the project - not only PHP developers.

**Comment by SamMousa** [comment]
Composer itself is just PHP so if you can run PHP code you can use composer to install dependencies.

I agree that frontend / backend should be separated.

**Comment by polak25** [comment]
+1

**Comment by vahidalvandi** [comment]
+1

**Comment by jafar-rezaei** [comment]
+1 any news about new version?

**Comment by johnshepherd** [comment]
+1

**Comment by Roman-1812** [comment]
+89

**Comment by liorphp** [comment]
> i think there is not a good solutions because you lose config parameters every update, if you know a method to prevent this i can consider to add

You don't.
because you can check if config array exist like on uploads.php .

You can declare that change on config array, will forced upgrade version.
When you change the config array the version will rise from X.1.8 ti X.1.2.0, and not X.1.8 ti X.1.9
So in composer we will write somethimg like this:
"trippo/responsive-filemanager": "X.1.*",

**Comment by lpotherat** [comment]
Does this "feature" will come one day ?
+1

**Comment by phpfui** [comment]
I think the best approach for composer compatibility is to break out the project into two separate projects.  I see @trippo has mentioned work on an update, but not sure where that stands.  Would be happy to help out here.

This is what I see needs to be done:

- Fully document the API's used.  This will allow for others to implement in another language.
- Full configuration of API endpoints called.  Defaults to current .php files for backward compatibility.
- Fully namespaced class library implementing each API end point.  Replacement  php files become simple autoloader and new class -> execute() type thing.  This class is obviously built up from parts.  This allows for simple implementations, but also more complex router applications.
 - Laravel and other framework plugins.  This would greatly help adoption (may already exist, I don't use a framework).
 - Rewrite would be backward compatible with existing implementation (ie. this repo now uses the two new repos to implement the same workflow).

I would be happy to do much of the PHP and API documentation work.  Just want buy in from @trippo as to not waste time with something that will not move forward.
